docs/API.md

```diff
@@ -0,0 +1,50 @@
+
+### Root - AsyncAPIDocument
+- `AsyncAPIDocument.hasContentType('<content type>')` : boolean (outcome of comment(s): [799481319](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-799481319))
+- `AsyncAPIDocument.channels()` : Channel[] (outcome of comment(s): [800282407](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800282407), [800935961](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800935961))
+- **(HELP NEEDED)** `AsyncAPIDocument.channelsSubscribes('<role>')` : Channel[] (outcome of comment(s): [800282407](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800282407), [800935961](https://github.com/asyncapi/shape-up-process/issues/84#issuecomment-800935961))
```

Comment:
The role is described in the [description of the API](https://github.com/asyncapi/parser-js/blob/5ac333dc96c4100a6e7f538581ce4c9d6bb889fc/docs/Intents.md#decisions) and might not be apparent to the user from the start. However, we took the decision to switch and add the role parameter to easy tool developer's interaction with the API. Especially for those tools which use the perspective of both.

It is one of those things that is not an easy decision to make 😅 I can understand if we want to keep it in the name of the intent.

Reply:
The problem with the param is that it introduces a new concept to learn so it increases friction. A developer using both "roles" can simply use ternary operators:

```
const value = x ? clientSubscribableChannels() : suscribingChannels()
```

Still easy for them.

Reply:
Yea it definitely will create some friction there, however, that friction will resolve itself in terms of code complexity and user interaction (This is just my opinion of course, and slightly biased). 

For example, how would the end user interact with a library that provides both perspectives? (If we think further ahead).

java-spring-template - uses `inverseOperations` as a boolean template parameter.
java-spring-cloud - uses `view` template parameter with the values `client` or `provider`.

At the moment it is up to developers to decide how to handle both views and by deciding on a concept for all tooling such as `role` it would create a common term that all toolings (that use this parser of course). Whether `role` is the correct wording for it that can be discussed 😄.

We might not want to do that, but it would be a huge benefit in my opinion.

Reply:
> The problem with the param is that it introduces a new concept to learn so it increases friction. A developer using both "roles" can simply use ternary operators:

We had mixed feelings about the `role` concept as well in this place. However, we moved forward because of the reasons you can find in [here](https://github.com/asyncapi/parser-js/pull/263/files#diff-b916c5a0628af4b4bdb634e7612d93408a30119addea9aab98fe5ac7c28adeafR37-R45).
I agree those are not written in stone (@jonaslagoni  wdyt?) so in this case, I agree that **explicit** is better than **implicit** here, and removing the concept of `role` as an argument in favor of generating all the different methods (one per message operation type) would give a self-explanatory API.

Reply:
> For example, how would the end user interact with a library that provides both perspectives? (If we think further ahead).

Looking at the future, this confusion should disappear. Most likely with something like [this](https://github.com/asyncapi/spec/issues/390#issuecomment-662032185).

Reply:
We @jonaslagoni and I decided to remove the role arg in favor of having methods that represent all the intents. Changes were added here https://github.com/asyncapi/parser-js/pull/263/commits/6da212318e582f0927685f25936d882ff6c529a5